src/main/java/io/odpf/firehose/factory/FirehoseConsumerFactory.java

```diff
@@ -86,8 +89,26 @@ public FirehoseConsumerFactory(KafkaConsumerConfig kafkaConsumerConfig, StatsDRe
      * @return FirehoseConsumer firehose consumer
      */
     public FirehoseConsumer buildConsumer() {
-
-        Filter filter = new MessageFilter(kafkaConsumerConfig, new Instrumentation(statsDReporter, MessageFilter.class));
+        FilterConfig filterConfig = ConfigFactory.create(FilterConfig.class, config);
+        Filter filter;
+        switch (filterConfig.getFilterEngine()) {
+            case JSON:
+                Instrumentation filterInstrumentation = new Instrumentation(statsDReporter, JsonFilter.class);
+                JsonFilter.logConfigs(filterConfig, filterInstrumentation);
+                try {
+                    JsonFilter.validateConfigs(filterConfig);
+                } catch (FilterException e) {
+                    instrumentation.logError("Failed to create filter due to invalid config");
```

Comment:
you can log this inside validateConfig(), no need to catch it here.
Also throw IllegalArgumentException (Which is Runtime) and not filterexception.. because filtering is not yet started.. this is checking of configs.

Reply:
fixed